client/state/data-layer/wpcom/checklist/index.js

```diff
@@ -7,16 +7,36 @@ import { get, noop } from 'lodash';
  * Internal dependencies
  */
 import { SITE_CHECKLIST_REQUEST, SITE_CHECKLIST_TASK_UPDATE } from 'state/action-types';
+import { SITE_CHECKLIST_KNOWN_TASKS } from 'my-sites/customer-home/cards/tasks/site-setup-list/get-task';
 import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
 import { http } from 'state/data-layer/wpcom-http/actions';
 import { receiveSiteChecklist } from 'state/checklist/actions';
 
 import { registerHandlers } from 'state/data-layer/handler-registry';
 
-// The checklist API requests use the http_envelope query param, however on
-// desktop the envelope is not being unpacked for some reason. This conversion
-// ensures the payload has been unpacked.
-const fromApi = ( payload ) => get( payload, 'body', payload );
+// Transform the response to a data / schema calypso understands, eg filter out unknown tasks
+const fromApi = ( payload ) => {
+	// The checklist API requests use the http_envelope query param, however on
+	// desktop the envelope is not being unpacked for some reason. This conversion
+	// ensures the payload has been unpacked.
+	const data = get( payload, 'body', payload );
+
+	if ( ! data ) {
+		throw new TypeError( `Missing 'body' property on API response` );
+	}
+	if ( ! Array.isArray( data.tasks ) ) {
```

Comment:
Here is the issue with the placeholder, tasks is not an Array, is an object

Reply:
Transferred here https://github.com/Automattic/wp-calypso/issues/44099